src/odemis/driver/zeiss.py

```diff
@@ -513,66 +560,101 @@ def _refreshPosition(self):
 
     def _doMoveRel(self, future, shift):
         """
-        move by the shift
-        shift (float): unit m
+        prepare the stage for a relative move using the values in shift
+        :param future: a future that can be used to manage a move
+        :param shift (dict): contains float position values in m
         """
-        x, y, z, _ = self.parent.GetStagePosition()
+        # set the values of these axes to 0.0
+        # if none of these are used in the move the command for 3 axes will be used
+        t = r = m = 0.0
+
+        # get the stage position for all axes
+        x, y, z, t, r, m, _ = self.parent.GetStagePosition6()
+
+        target_pos = {}
+
         if "x" in shift:
             x += shift["x"] * 1e3
+            target_pos["x"] = x * 1e-3
         if "y" in shift:
             y += shift["y"] * 1e3
+            target_pos["y"] = y * 1e-3
         if "z" in shift:
             z += shift["z"] * 1e3
-
-        target_pos = self._applyInversion({"x": x * 1e-3, "y": y * 1e-3, "z": z * 1e-3})
-        # Check range (for the axes we are moving)
+            target_pos["z"] = z * 1e-3
+        if "rx" in shift:
+            t += shift["rx"]
+            target_pos["rx"] = t
+        if "rz" in shift:
+            r += shift["rz"]
+            target_pos["rz"] = r
```

Comment:
Convert to rad, and then back to degree for target_pos.
I think it might help/clarify if we had two functions to convert a pos (dict) from/to "user view" (ie, m/rad with inversion) and "hw view" (ie, mm/deg without inversion). Something like `pos_to_user()` and `pos_to_hw()`.

Reply:
There is no degree/rad conversion necessary anymore.